src/actions/actions.ts

```diff
@@ -583,6 +589,21 @@ export class DeleteOperatorXVisual extends BaseOperator {
 }
 
 @RegisterAction
+export class UpperCaseOperatorXVisual extends BaseOperator {
+    public keys = ["U"];
+    public modes = [ModeName.Visual, ModeName.VisualLine];
+    
+    public async run(vimState: VimState, start: Position, end: Position): Promise<VimState> {
+      [start, end] = adjustVimSelectionPosition(vimState, start, end);
```

Comment:
Will do that. Once the code is updated you can know if I fully get your ideas :)

BTW, send you an email about other funny stuff.